### PR TITLE
Issue 10094

### LINKED ISSUE #10094: [Coverity CID :188169] Null pointer dereferences in /subsys/net/ip/rpl.c
Static code scan issues seen in File: /subsys/net/ip/rpl.c
 Category: Null pointer dereferences
 Function: dio_timer
 Component: Networking
 CID: 188169
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by satyacube**
subsys: net: Null pointer dereferences

Check for NULL IPv6 addr values from net_if_ipv6_get_ll() in rpl.c
If NULL, print a debug statement

Fixes #10094.

Signed-off-by: Satya Bhattacharya <satyacube@gmail.com>

### CODE REVIEW

File: subsys/net/ip/rpl.c [hunk 1]
```diff
@@ -635,12 +635,16 @@ static void dio_timer(struct k_work *work)
...
 						   NET_ADDR_PREFERRED);
 
-			net_rpl_dio_send(rpl_default_iface, instance, addr,
-					 NULL);
-
+			if (addr) {
+				net_rpl_dio_send(rpl_default_iface, instance, addr,
+					         NULL);
 #if defined(CONFIG_NET_STATISTICS_RPL)
 			instance->dio_send_pkt++;
```
Comment (jfischer-phytec-iot) [inline subsys/net/ip/rpl.c]: please indent this line

File: subsys/net/ip/rpl.c [hunk 2]
```diff
@@ -635,12 +635,16 @@ static void dio_timer(struct k_work *work)
...
 #endif
+			} else {
+				NET_DBG("Sending DIO failed, IPv6 link local "
+				        "address is NULL");
```
Comment (jfischer-phytec-iot) [inline subsys/net/ip/rpl.c]: Is it ok to resubmit with `instance->dio_next_delay` parameter in this case?

**Review by jukkar — CHANGES_REQUESTED** [review]

File: subsys/net/ip/rpl.c (hunk 2 above)
Comment (satyacube) [reply subsys/net/ip/rpl.c]: IF addr is NULL should we?

`			k_delayed_work_submit(&instance->dio_timer,
					      DIO_TIMEOUT);
			return;`

**Comment by jukkar** [comment]
Not sure what happened here and why this PR was closed, your patch was almost there.

**Pull request closed without merging**